index.d.ts

```diff
@@ -1,16 +1,23 @@
 import { AxiosRequestConfig, AxiosError, AxiosPromise } from 'axios'
 
 interface ResponseValues {
-  data: any;
-  loading: boolean;
-  error?: AxiosError;
+  data: any
+  loading: boolean
+  error?: AxiosError
 }
 
 interface Options {
   manual: boolean
 }
 
+interface RefetchOptions {
+  useCache: boolean
```

Comment:
As discussed in the Issue, I think it would be better to allow a user to choose to use caching for the standard useAxios behaviour or refetch. If that was decision was made, useCache could move to the Options interface. As it stands though, having useCache in this new RefetchOptions interface works.

Reply:
I'm not convinced we should have that option there to be honest. I mean the hook will execute whenever the component renders, meaning that having that option there and choosing to not use the cache will trigger a request whenever the component re-renders, which I see no reason why anybody would want to do. I think we should stick to the option being provided to the "refetch" function, with cache off by default, as in this PR.

Reply:
"I see no reason why anybody would want to do" - I guess it depends how volatile the data is on the server. TBH I am happy with what you have proposed so I reckon you should go ahead and merge the PR.

Reply:
Not really, because you shouldn't rely on the frequency with which a component renders to carry out any application logic. That should be transparent for the developer.

Reply:
Ok, fair enough. It's just that cache invalidation is always a nightmare! Good work @simoneb

Reply:
I know. If you've used Apollo you know that it's really not easy to implement and use.